CONTRIBUTING.md

```diff
@@ -0,0 +1,61 @@
+
+- Unclear documentation
+- Bugs, crashes
+- Enhancement ideas
+- ... and more
+
+Please try to be as descriptive as possible. Provide as much information as you are able to (e.g. browser version, operating system, SDK version). If you are seeing an error, paste as much of the log as you can.
+
+Of course, after submitting an Issue you are free to tackle the problem yourself (see "Submitting changes" below), and encouraged to do so.
+
+## Choosing an Issue
+
+If you want to contribute code but don't know what to work on, we try to always keep some Issues marked [`help-wanted`](https://github.com/NebulousLabs/skynet-js/labels/help%20wanted) or [`good-first-issue`](https://github.com/NebulousLabs/skynet-js/labels/good%20first%20issue) We want everyone to be able to contribute!
+
+## Development
+
+We follow the common development process in the company. We use [git](https://git-scm.com/) as our [version control system](https://en.wikipedia.org/wiki/Version_control) (VCS). We develop new features in separate git branches, raise [Pull Requests](https://help.github.com/en/articles/about-pull-requests), put them under peer review, and we merge them only after they pass the QA checks and [continuous integration](https://en.wikipedia.org/wiki/Continuous_integration) (CI). We never commit directly to the `master` branch.
+
+For useful resources, please see:
+
+- [Git basics](https://git-scm.com/book/en/v1/Getting-Started-Git-Basics) for Git beginners
+- [Git best practices](https://sethrobertson.github.io/GitBestPractices/)
+
+## Submitting changes (Pull Requests)
+
+If you are a complete newbie, click [here](https://github.com/firstcontributions/first-contributions) for an easy-to-follow guide (with pictures!)
+
+We follow the standard procedure for submitting Pull Requests. Please refer to the [official GitHub documentation](https://help.github.com/articles/creating-a-pull-request/) if you are unfamiliar with the procedure. If you still need help, we are more than happy to guide you along!
+
+**Note:** Before submitting code, please either choose an existing Issue, or write your own, describing the problem you are solving. We would like for every PR to have an accompanying Issue, so that we know concretely what the problem is and can track its resolution.
+
+### Code Style
+
+We try to follow standard best practices for Javascript and Typescript. These best practices are enforced by CI (see below) and code review (also see below).
+
+### Format
+
+We don't have any hard-and-fast rules for styling git commits or branch names, but try to look at existing PRs first to get a sense of how they are structured.
+
+### Running tests (CI script)
+
+Submitted PRs are expected to pass continuous integration (CI), which, among other things, runs a test suite on your PR to make sure that your code doesn't have any bugs.
+
```

Comment:
Maybe a comment here indicating that any new methods need accompanying tests, and if you aren't familiar with Jest, make your PR and we can help from there?

(This might be too specific to the repo if it's company wide.)

Reply:
Sure, I'll try to word it in a generic way, and make the rest of the document a bit more generic